core/src/main/java/org/bitcoinj/crypto/HDPath.java

```diff
@@ -162,6 +162,20 @@ public HDFullPath parent() {
         public HDPartialPath asPartial() {
             return new HDPartialPath(this.childNumbers);
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if ((o == null) || getClass() != o.getClass()) return false;
+            HDFullPath other = (HDFullPath) o;
+            return Objects.equals(this.hasPrivateKey, other.hasPrivateKey) &&
+                    super.equals(other);
+        }
```

Comment:
Can't this be shortened to:
```
            if (!super.equals(o)) return false;
            HDFullPath other = (HDFullPath) o;
            return Objects.equals(this.hasPrivateKey, other.hasPrivateKey);
```

Reply:
Not sure, I'm following a pattern here that we established years ago – I think based on "Java best practices".

I would suggest that if we'd like to change this pattern, we should do it for the entire codebase and hence it would be out of scope for this PR.

Reply:
The thing I like about the shortened version is hit makes it more clear that it's the same as `super` but adds the comparison of `hasPrivateKey`. Not a big deal, though.

Reply:
> I think based on "Java best practices"

For reference: https://github.com/bitcoinj/bitcoinj/pull/65